seatunnel-core/seatunnel-core-base/src/main/java/org/apache/seatunnel/config/ConfigBuilder.java

```diff
@@ -102,6 +103,22 @@ private boolean checkIsStreaming() {
         return sourceConfigList.get(0).getString(PLUGIN_NAME_KEY).toLowerCase().endsWith("stream");
     }
 
+    private boolean checkIsContainHive() {
+        List<? extends Config> sourceConfigList = config.getConfigList(PluginType.SOURCE.getType());
+        for (Config c : sourceConfigList) {
+            if (c.getString(PLUGIN_NAME_KEY).toLowerCase().contains("hive")) {
+                return true;
+            }
+        }
+        List<? extends Config> sinkConfigList = config.getConfigList(PluginType.SINK.getType());
+        for (Config c : sinkConfigList) {
+            if (c.getString(PLUGIN_NAME_KEY).toLowerCase().contains("hive")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
```

Comment:
Overall LGTM, just leave a question, is it appropriate to put this kind of business-related here?

Reply:
I think this method is similar with checkIsStreaming(), both will change behavior at the framework level. But hive is so special, any better idea?

Reply:
> I also agree don't put this logic in ConfigBuilder, since you still need to use this result in `SparkEnvironment`.
> 
> It might be better to just modified code in `SparkEnvironment` and remove the change of ConfigBuilder.


In flink, this config also need, but not support hive in flink at now

Reply:
> > I also agree don't put this logic in ConfigBuilder, since you still need to use this result in `SparkEnvironment`.
> > It might be better to just modified code in `SparkEnvironment` and remove the change of ConfigBuilder.
> 
> In flink, this config also need, but not support hive in flink at now

If so, it's ok to add a method here, anyway we can still refactor code in the future.